src/ceph-daemon/ceph-daemon

```diff
@@ -436,6 +503,81 @@ def create_daemon_dirs(fsid, daemon_type, daemon_id, uid, gid,
             os.fchown(f.fileno(), uid, gid)
             f.write(keyring)
 
+    if daemon_type == 'prometheus':
+        
+        # Set up directories
+        data_dir_root = get_data_dir(fsid, daemon_type, daemon_id)
+        makedirs(os.path.join(data_dir_root,'etc/prometheus'), uid, gid, 0o755)
+        makedirs(os.path.join(data_dir_root,'data'), uid, gid, 0o755)
+
+        config = get_parm(args.config_json)
+        if not config:
+            raise Error("Prometheus deployment requires config-json to be set")
+
+        prometheus_yml = dict()
+
+        if "prometheus.yml" in config:
+            # trust the provided version is appropriate
+            prometheus_yml = config.get('prometheus.yml')
+            logger.debug("Using supplied prometheus.yml definition")
+        else:
+            if "mgrs" in config:
+                prometheus_yml = Monitoring.components['prometheus']['config']['prometheus.yml']
+                if isinstance(config['mgrs'], list):
+                    targets=[]
+                    for mgr in config['mgrs']:
+                        targets.append('{}:9283'.format(mgr))
+                    prometheus_yml['scrape_configs'].append({
+                            "job_name": "ceph",
+                            "static_configs": [
+                                { "targets": targets,
+                                  "labels": {
+                                    "instance": "ceph_cluster"
+                                    }
+                                }
+                            ],
+                        })
+                    logger.debug("Using default prometheus settings with supplied list of ceph mgrs")
+                else:
+                    raise Error("config-json supplied with invalid mgrs setting - must be a list")
+            else:
+                raise Error("config-json must either specify a valid prometheus.yml, or a mgrs list")
+
+        # create configs
+        with open(os.path.join(data_dir_root,'etc/prometheus/prometheus.yml'), 'w') as f:
+            os.fchown(f.fileno(), uid, gid)
+            os.fchmod(f.fileno(), 0o600)
+            f.write(yaml.safe_dump(prometheus_yml, explicit_start=True, indent=2))
+
+def get_parm(option):
+    """Read requested arg, and return as a dict"""
+
+    if not option:
+        return None
+
+    if option == '-':
+        try:
+            j = injected_stdin
+        except NameError:
+            j = sys.stdin.read()
+    else:
+        # inline json string
+        if option[0] == '{' and option[-1] == '}':
```

Comment:
I can't see any code in this PR that make use inline JSON for arguments. Can we remove this?

Reply:
see comment above about get_parm becoming the generic json parser